service.upnext/resources/lib/script.py

```diff
@@ -88,12 +88,13 @@ def test_popup(window):
     popup.show()
     step = 0
     wait = 100
+    wait_s = wait / 1000
     timeout = 10000
     monitor = Monitor()
     while popup and step < timeout and not monitor.abortRequested():
         if popup.pause:
             continue
-        sleep(wait)
+        monitor.waitForAbort(wait_s)
```

Comment:
Any reason for this change? Sleeping for 0.1 seconds doesn't necessarily require you to use waitforabort

Reply:
cc @MoojMidge

Reply:
It was an attempt to prevent Kodi from hanging when the test popup was created and the addon was restarted or Kodi was shutdown. 

Couldn't figure out why a 0.1s sleep would cause the hang, but decided to try waitForAbort in case there was some interaction between sleep and abortRequested that was causing the problem. 

Turns out the actual problem was that Kodi wasn't stopping scripts on shutdown and so the abortRequested call wasn't actually doing anything, which is fixed in https://github.com/xbmc/xbmc/pull/18305

Retained the call to waitForAbort as it seemed slightly more idiomatic to only use the Monitor methods going forward.